### PR TITLE
Show or create direct message room

### PR DESCRIPTION
**Pull request by Florian14**
Part of #96 

Notes:
- create DM action has been disabled because there are issues with the account data which is not synced correctly
- retrieve DM is still enabled but does not really work for the same reason

Content:

- Create a DM with a user if there is no existing one
- Open the existing DM if any
- Show an error dialog with a retry button in case of an error when creating the room

As the search is not implemented yet, we're only able to start & retrieve a chat with a matrix ID

### CODE REVIEW

**Comment by Florian14** [comment]
> I have a comment about the `ErrorDialog`. Also, I'm sorry but I'm not sure what I should be testing.
> 
> Does only opening an existing DM work, or is there something else that I can try? I tried creating a room with an existing user, but the UI just got stuck.

This PR is not really ready to be tested since there are issues with the account data on the rust side... I have unplugged the DM creation so you can only try to retrieve DMs but most of the time the account data is empty and you cannot retrieve any DM. If you plug again the DM creation, you should be able to create a DM (should be tested on a test account because it will override the account data) and then retrieve this DM.

I tried recording a demo but the result was not great because of the account data issues. :/ I made this PR reviewable because I don't have more work to do on it and my other task is at the top of this one.

File: features/createroom/impl/src/main/kotlin/io/element/android/features/createroom/impl/root/CreateRoomRootView.kt [hunk 1]
```diff
@@ -83,6 +103,22 @@
             }
         }
     }
+
+    when (state.startDmAction) {
+        is Async.Loading -> {
+            ProgressDialog(text = stringResource(id = StringR.string.common_creating_room))
+        }
+        is Async.Failure -> {
+            ErrorDialog(
```
[the remark this answers is not in the document]
Comment (jmartinesp) [reply features/createroom/impl/src/main/kotlin/io/element/android/features/createroom/impl/root/CreateRoomRootView.kt:112]: Maybe a `RetryDialog` based on either the `ConfirmationDialog` or `ErrorDialogs` so it has some default values?

**Comment by jmartinesp** [comment]
> I made this PR reviewable because I don't have more work to do on it and my other task is at the top of this one.

Should we merge it as is then and then we'll come back to it in another task?

**Comment by jmartinesp** [comment]
I can't get "Open the existing DM if any" to work on my account. Is there some special way to trigger it?

**Comment by Florian14** [comment]
> I can't get "Open the existing DM if any" to work on my account. Is there some special way to trigger it?

Let's consider the code only for now, I was able to get it working by changing the SS view... but this is very hacky :/
We'll validate the functional part when I will plug again the DM creation

File: features/createroom/impl/src/main/kotlin/io/element/android/features/createroom/impl/root/CreateRoomRootPresenter.kt [hunk 2]
```diff
@@ -43,20 +53,41 @@ class CreateRoomRootPresenter @Inject constructor(
...
+        }
+
         fun handleEvents(event: CreateRoomRootEvents) {
             when (event) {
-                is CreateRoomRootEvents.StartDM -> handleStartDM(event.matrixUser)
+                is CreateRoomRootEvents.StartDM -> startDm(event.matrixUser)
+                CreateRoomRootEvents.RetryStartDM -> {
+                    startDmAction.value = Async.Uninitialized
+                    userListState.selectedUsers.firstOrNull()?.let { startDm(it) }
+                }
```
[the remark this answers is not in the document]
Comment (Florian14) [reply features/createroom/impl/src/main/kotlin/io/element/android/features/createroom/impl/root/CreateRoomRootPresenter.kt]: It was to ensure the action is correctly reset even if there is no selected user (should not happen...). I agree this is a bit overkill, I took your suggestion into account, reuse the `startDM` action and just trigger `CancelStartDM` if there is no selected user

**Pull request merged**